taqueria-plugin-contract-types/tasks.ts

```diff
@@ -0,0 +1,128 @@
+import { SanitizedArgs, ActionResponse, Failure, LikeAPromise, ProxyAction } from "@taqueria/node-sdk/types";
+import glob from 'fast-glob'
+import { join } from 'path'
+import { generateContractTypesProcessContractFiles } from "./src/cli-process";
+
+type PluginOpts = {
+    // TODO: Document these
+    typescriptDir: string,
+    typeAliasMode?: 'local' | 'file' | 'library' | 'simple',
+};
+type Opts = SanitizedArgs & Record<string, unknown>;
+
+const getContractAbspath = (contractFilename: string, parsedArgs: Opts) => 
+    join(parsedArgs.artifactsDir, /\.tz$/.test(contractFilename) ? contractFilename : `${contractFilename}.tz`)
+
+
+const generateContractTypes = (parsedArgs: Opts & PluginOpts) => async (contractFilename: string) : Promise<string> => {
+    const contractAbspath = getContractAbspath(contractFilename, parsedArgs);
+    await generateContractTypesProcessContractFiles({
+        inputTzContractDirectory: parsedArgs.artifactsDir,
+        inputFiles: [contractAbspath],
+        outputTypescriptDirectory: parsedArgs.typescriptDir,
+        format: 'tz',
+        typeAliasMode: parsedArgs.typeAliasMode ?? 'file',
+    });
+
+    return `${contractFilename}: Types generated`;
+
+    // TODO: Generate contract michelson
```

Comment:
@rick could we supplement these todos please with issue #s?

Reply:
![giphy (20)](https://user-images.githubusercontent.com/6259/153921125-8fe757d3-8ed5-4ba2-ae57-5317fe4cbda3.gif)

cc https://github.com/rick/you-rang/issues/1